### PR TITLE
Completed User functionality - Create account, delete account, update account, sign out, get users by name, id, etc.

### PR DESCRIPTION
**Pull request by b1gd3vd0g**
Everything works fine. Front and back end done. It's not super pretty, but we'll fix that as time allows.

### CODE REVIEW

File: backend/mongoose/functions/user_functions.js [hunk 1]
```diff
@@ -119,6 +119,31 @@ const getUserByUsername = async (un) => {
...
+        code: 200,
+        user: user
+      };
+    }
+  } catch (error) {
+    return {
+      success: false,
+      error: error.message,
+      code: 500
+    };
```
Comment (tclaar) [inline backend/mongoose/functions/user_functions.js:143]: We're also catching errors in the `user` service itself, right? I removed this extra try-catch from my service functions and they seem to handle errors alright.

File: frontend/src/components/logged_in/DeleteAcc.js [hunk 2]
```diff
@@ -6,20 +6,29 @@ import { requestAccountDeletion } from "../../requests/user_requests";
...
       <input type='password' className='Password' />
       <br />
+      <br />
       <button className='btn btn-lg btn-danger' onClick={async () => {
         const pw = document.querySelector('.DeleteAcc .Password').value.trim();
-        requestAccountDeletion({ un: userState.user.username, pw: pw });
-        setUserState({
-          loggedIn: false
-        });
+        const deletion = await requestAccountDeletion({ un: userState.user.username, pw: pw });
```
Comment (tclaar) [inline frontend/src/components/logged_in/DeleteAcc.js:19]: I should start doing something like this for my HTTP requests, it's way cleaner than directly calling fetch here.

**Review by tclaar — APPROVED** [review]
Nice work!

File: backend/mongoose/functions/user_functions.js (hunk 1 above)
Comment (b1gd3vd0g) [reply backend/mongoose/functions/user_functions.js:143]: Yes, that's correct.  I included it as kinda a failsafe, there was a point or two where it was useful, but you're right.

File: frontend/src/components/logged_in/DeleteAcc.js (hunk 2 above)
Comment (b1gd3vd0g) [reply frontend/src/components/logged_in/DeleteAcc.js:19]: Modularity ftw

**Pull request merged**